From 010f198e4ca61df8421a6db41e65e551d5a61b0a Mon Sep 17 00:00:00 2001
From: Kamil Debski <k.debski@samsung.com>
Date: Fri, 25 Jan 2013 06:29:56 -0300
Subject: [PATCH 0877/1083] [media] v4l: Define video buffer flag for the COPY
 timestamp type

Define video buffer flag for the COPY timestamp. In this case the timestamp
value is copied from the OUTPUT to the corresponding CAPTURE buffer.

Signed-off-by: Kamil Debski <k.debski@samsung.com>
Signed-off-by: Kyungmin Park <kyungmin.park@samsung.com>
Reviewed-by: Sylwester Nawrocki <s.nawrocki@samsung.com>
Acked-by: Hans Verkuil <hans.verkuil@cisco.com>
Signed-off-by: Sylwester Nawrocki <s.nawrocki@samsung.com>
Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>
(cherry picked from commit 53bf0f446bc387eabdd535dca080789cc74607f4)

Conflicts:
	Documentation/DocBook/media/v4l/io.xml
	include/uapi/linux/videodev2.h
	include/linux/videodev2.h

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
(cherry picked from commit 80f7bc5a6fe616f8ca906199cefa047e0ee83998)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 Documentation/DocBook/media/v4l/io.xml | 4 ++++
 include/linux/videodev2.h              | 1 +
 2 files changed, 5 insertions(+)

diff --git a/Documentation/DocBook/media/v4l/io.xml b/Documentation/DocBook/media/v4l/io.xml
index 8f885b9..3495052 100644
--- a/Documentation/DocBook/media/v4l/io.xml
+++ b/Documentation/DocBook/media/v4l/io.xml
@@ -984,6 +984,10 @@ in which case caches have not been used.</entry>
 	    <constant>CLOCK_MONOTONIC</constant> clock. To access the
 	    same clock outside V4L2, use
 	    <function>clock_gettime(2)</function> .</entry>
+	    <entry><constant>V4L2_BUF_FLAG_TIMESTAMP_COPY</constant></entry>
+	    <entry>0x4000</entry>
+	    <entry>The CAPTURE buffer timestamp has been taken from the
+	    corresponding OUTPUT buffer. This flag applies only to mem2mem devices.</entry>
 	  </row>
 	</tbody>
       </tgroup>
diff --git a/include/linux/videodev2.h b/include/linux/videodev2.h
index e0b0440..e0b2626 100644
--- a/include/linux/videodev2.h
+++ b/include/linux/videodev2.h
@@ -681,6 +681,7 @@ struct v4l2_buffer {
 #define V4L2_BUF_FLAG_TIMESTAMP_MASK		0xe000
 #define V4L2_BUF_FLAG_TIMESTAMP_UNKNOWN		0x0000
 #define V4L2_BUF_FLAG_TIMESTAMP_MONOTONIC	0x2000
+#define V4L2_BUF_FLAG_TIMESTAMP_COPY		0x4000
 
 /*
  *	O V E R L A Y   P R E V I E W
-- 
1.8.3.2

